### PR TITLE
Simplify the records iterator

### PR DESCRIPTION
**Pull request by Ladas**
Simplify the iterator and allow it to run pure sql fetch (which will cause speedup). Make sure specs with this pass with serialized format too (there was a timestamp comparing change needed to pass all specs, since the pure SQL doesn't cast the timestamp for us)

### CODE REVIEW

File: lib/inventory_refresh/application_record_iterator.rb [hunk 1]
```diff
@@ -2,19 +2,11 @@ module InventoryRefresh
   class ApplicationRecordIterator
     attr_reader :inventory_collection, :manager_uuids_set, :iterator, :query
```
Comment (agrare) [inline lib/inventory_refresh/application_record_iterator.rb]: @Ladas can `@manager_uuids_set` be deleted now?

Comment (agrare) [reply lib/inventory_refresh/application_record_iterator.rb]: Actually are any of these ivars still used?

Comment (Ladas) [reply lib/inventory_refresh/application_record_iterator.rb]: oh yeah

Comment (Ladas) [reply lib/inventory_refresh/application_record_iterator.rb]: done

**Pull request merged**